scipy/spatial/transform/_rotation.pyx

```diff
@@ -2766,33 +2806,65 @@ cdef class Rotation:
 
         where :math:`w_i`'s are the `weights` corresponding to each vector.
 
-        The rotation is estimated with Kabsch algorithm [1]_.
+        The rotation is estimated with Kabsch algorithm [1]_, and solves what
+        is known as the "pointing problem", or "Wahba's problem".
+
+        There are two special cases. The first is when one of the weights is
+        infinite. In this case, the rotation is estimated by first applying the
+        Kabsch algorithm to the subset of vectors with finite weights. The
+        rotation is then found from that result, to the infinite-weight vectors
+        via the "align-constrain" algorithm [2]_. If `a` and `b` are both shape
+        (2, 3) arrays, then the initial Kabsch algorithm step is skipped and
+        the align-constrain algorithm is used directly.
+
+        The align-constrain algorithm works by first creating orthonormal
+        frames A and B by applying the Gram-Schmidt [3]_ process to each of the
+        sets of shape (2, 3) vectors `a` and `b`. This will keep primary axes
+        A[0] and B[0] unchanged from `a[0]` and `b[0]`, while the secondary
+        axes A[1] and B[1] are formed by keeping only the component of the
+        secondary vectors that is normal to the primary. The rotation from
+        frame B to A is then calculated.
+
+        The second special case is if a single vector is given for `a` and `b`,
+        in which the shortest distance rotation that aligns `b` to `a` is
+        returned. Any input for `weights` is ignored in this case. This is also
+        the behavior if `a[0]` and `a[1]` or `b[0]` and `b[1]` as found in the
```

Comment:
Why do you think it's beneficial to consider two vectors as a special case for "constrain" algorithm, especially with tolerance? What if we get 10 almost parallel vectors? 

From the implementation perspective I don't like the fact that we need to implement the whole another function to support it, whereas it should be handled correctly with a general algorithm as far as I understand.

Reply:
Now I've read the code, I understand what is your approach here.

Now need to understand why it works.